velocity/src/main/java/me/confuser/banmanager/webenhancer/velocity/VelocityPlugin.java

```diff
@@ -0,0 +1,161 @@
+                        e.printStackTrace();
+                        return;
+                }
+
+                setupRunnables();
+                setupListeners();
+                setupCommands();
+
+        }
+
+        private PluginInfo setupConfigs() throws IOException {
+                if (!dataDirectory.exists()) {
+                        dataDirectory.mkdir();
+                }
+
+                for (String name : configs) {
+                        File file = new File(dataDirectory, name);
+
+                        if (!file.exists()) {
+                                try (InputStream in = getResourceAsStream(name)) {
+                                        Files.copy(in, file.toPath());
+                                } catch (IOException e) {
+                                        e.printStackTrace();
+                                }
+                        } else {
+                                Reader defConfigStream = new InputStreamReader(getResourceAsStream(file.getName()), "UTF8");
+
+                                YamlConfiguration conf = YamlConfiguration.loadConfiguration(file);
+                                YamlConfiguration defConfig = YamlConfiguration.loadConfiguration(defConfigStream);
+                                conf.setDefaults(defConfig);
+                                conf.options().copyDefaults(true);
+                                conf.save(file);
+                        }
+                }
+
+                // Load plugin.yml
+                PluginInfo pluginInfo = new PluginInfo();
+                Reader defConfigStream = new InputStreamReader(getResourceAsStream("plugin.yml"), "UTF8");
+                YamlConfiguration conf = YamlConfiguration.loadConfiguration(defConfigStream);
+                ConfigurationSection commands = conf.getConfigurationSection("commands");
+
+                for (String command : commands.getKeys(false)) {
+                        ConfigurationSection cmd = commands.getConfigurationSection(command);
+
+                        pluginInfo.setCommand(new PluginInfo.CommandInfo(command, cmd.getString("permission"), cmd.getString("usage"), cmd.getStringList("aliases")));
+                }
+
+                return pluginInfo;
+        }
+
+        @SneakyThrows
+        private InputStream getResourceAsStream(String resource) {
+                Class<?> cls = getClass();
+                ClassLoader cLoader = cls.getClassLoader();
+                return cLoader.getResourceAsStream(resource);
+        }
+
+        private void setupCommands() {
+                for (CommonCommand cmd : plugin.getCommands()) {
+                        new VelocityCommand(cmd, this);
+                }
+
+                logger.info("Registered commands");
+        }
+
+        public void setupRunnables() {
+//    Runner syncRunner = new Runner(new ExpiresSync(plugin));
+
+//    getServer().getScheduler().runTaskTimerAsynchronously(this, syncRunner, 10L, 10L);
```

Comment:
Should this not be uncommented?

Reply:
In this case, I only copied the bungee and bukkit code. It's also commented out there, even if I didn't quite understand why. Made the duplicate in case you want to work on this, but I can delete it if you want to,